server/src/service.rs

```diff
@@ -123,46 +125,85 @@ impl Service {
 }
 
 impl Service {
-    pub async fn run() -> Result<()> {
+    pub async fn run() -> Result<zbus::Connection> {
```

Comment:
Instead of making this return the connection, i think it makes more sense to create it in main.rs and pass it as a param here.